compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -333,6 +333,300 @@ bool OMR::CFGSimplifier::simplifyInstanceOfTestToCheckcast(bool needToDuplicateT
    return true;
    }
 
+// Looks for one of the patterns of the form:
+//   Pattern-1:
+//    ificmplt (i, 0) --------
+//        | (FallThrough)     |
+//    ificmplt (i, length) ======    
+//        |                   |  |
+//      throw  <--------------   |
+//        |                      |
+//      return <=================
+//
+//   Pattern-2:
+//    ificmplt (i, 0) -----------
+//        | (FallThrough)        |
+//    ificmpge (i, length) ------|   
+//        |                      |
+//      return                   |
+//        |                      |
+//       throw <-----------------
+//
+//   Pattern-3:
+//    ificmpge (i, 0) -----------
+//        | (FallThrough)        |
+//      throw                    |
```

Comment:
I am not sure we recognize this pattern, but maybe I somehow missed it. I thought we check that the second "ifcmp" immediately follows the first "ifcmp".

Reply:
Yes, we match with this pattern. The second comparison is chosen based on the current comparison. For example, if the first compare is ificmplt i,0; then second compare immediately follows it. If the first comparison is ificmpge i,0; the second comparison is the branch block of the first compare.